### PR TITLE
ASRG Appliances

### PR DESCRIPTION
**Pull request by anoobindisguise**
#### Summary
Content "ASRG Appliances"

#### Purpose of change
Advanced Stirling Radioisotope Generators (ASRGs) can be found in game, these are a real world NASA creation that the military took and armored up, then hauled around to various military sites as seen on the roadside riot turret checkpoints and various military installations. Since these are clearly portable the survivor should be able to pick them up and repurpose them.

#### Describe the solution
https://rps.nasa.gov/system/downloadable_items/36_APP_ASRG_Fact_Sheet_v3_9-3-13.pdf

Rather than disassemble the shielding the player now turns the ASRG into an item, or can connect it to the grid on the spot. They can then haul them around (very bulky and heavy) and set them up for perpetual power. Per Nasa's design specs these produce 130w continuously. The equivalent of about 8-10 solar panels since these operate continuously irrespective of weather and sun radiance. The player can also disassemble the item into the base ASRG and get the shielding components to massively cut down the weight and volume, but it becomes radioactive in the process.

#### Describe alternatives you've considered
More power generation. These are a pretty big brick of shielding and the NASA version uses 1.2kg of pu-238, but the ones in game drop up to 3kg of plutonium when smashed. So it could be 2 or even 3 individual units of ASRGs contained in the same housing multiplying power output by that amount. More than willing to change this but as a start I'm going with the NASA specs.

Less radiation. These are not nuclear reactors, from what I have seen these are relatively safe even without all the big heavy military shielding and the pu-238 used is unlikely to be a toxic inhalant to humans. Thus this could also change.

Make these non-appliances able to go in a vehicle. Not efficient as they are very heavy but they're not so big that they couldn't fit. Right now though they're just appliances, as I'd initially like to make as few leaps of logic as possible.

Make accumulating too many reactors in one spot cause leech plants to spawn in your base (just kidding lol)

#### Testing
Tried these out in game, tried both connecting the ASRG to the grid immediately and disassembling it to get the base item and placing it elsewhere. Then hooked it up to a grid battery with an extension cord in both cases and got it working. I ran into a single bug, when I placed the ASRG in the world while holding it in my hands I got "phantom weight" that did not go away continuously knocking me down and overburdening me until I disassembled it and then dropped the item from my hands, although I did not try putting something else in my empty hands. This is likely a separate bug inherent to the appliance system.

#### Additional context
This will be partially dependent on #58362. 
Currently LEAK_DAM and LEAK_ALWAYS vehicle parts do not work. I copied their flags over from the Atomic light vehicle parts assuming precedent but it turns out that flag doesn't actually work on vehicle parts. This is a separate issue that will need fixing.

### CODE REVIEW

**Comment by AtomicFox556** [comment]
Plutonium-238 actually emits alpha particles as it decays, which are very easily blocked (even by a sheet of paper), so even very light plating would suffice if you don't expect anyone to try and break open the unit, as long as the material itself is sealed away.

That said, it will eventually decay into elements that emit (somewhat) more penetrative beta particles, but the immediate decay product of plutonium-238 (with its own half-life of 87 years) is uranium-234 with a half-life of 246 thousand years which likewise undergoes alpha decay, so I don't think it'd matter in game.

File: data/json/items/appliances.json [hunk 1]
```diff
@@ -320,5 +320,43 @@
...
+  {
+    "type": "GENERIC",
+    "id": "compact_ASRG",
+    "name": { "str_sp": "Unshielded Compact Advanced Sterling Radioisotope Generator" },
+    "looks_like": "t_plut_generator",
+    "description": "A moderately heavy brick of metal housing an ASRG.  Most of the lead and steel plating has been salvaged off so it's a lot less bulky and heavy.  However it's a lot less sturdy, so the threat of a radioactive breach is much more real.",
+    "weight": "60 kg",
+    "to_hit": -10,
+    "material": [ "lead", "steel" ],
+    "volume": "75 L",
```
Comment (AtomicFox556) [inline data/json/items/appliances.json:352]: The volume could well be even larger actually; even the listed prototype dimensions give you 136.3 L, assuming it's a rectangular cuboid.

**Comment by anoobindisguise** [comment]
> Plutonium-238 actually emits alpha particles as it decays, which are very easily blocked (even by a sheet of paper), so even very light plating would suffice if you don't expect anyone to try and break open the unit, as long as the material itself is sealed away. Notably, in 1970s a small batch of implanted cardiac pacemakers with a small slug of Pu-238 inside them to power the device was made.
> 
> That said, it will eventually decay into elements that emit (somewhat) more penetrative beta particles, but the immediate decay product of plutonium-238 (with its own half-life of 87 years) is uranium-234 with a half-life of 246 thousand years which likewise undergoes alpha decay, so I don't think it'd matter in game.
> 
> As long as they actually come with some kind of steel plating, I'd make them emit dangerous radiation only if they were damaged at least enough so as to make them stop working.
> 
> Wikipedia lists the weight of such a prototype ASRG unit at 32 kg and dimensions at 76 cm x 46 cm x 39 cm; I guess if you also include steel plating and use components that may be heavier/bulkier (don't need to care as much about this if you're not planning to send this into space), then weight of 60 kg and dimensions of about 90 cm x 55 cm x 45 cm could be reasonable.


Thank you! Super helpful info, I appreciate it. Applied your suggestions.

**Comment by PatrikLundell** [comment]
Alpha radiation isn't particularly dangerous outside of the body, but it's dangerous inside of it (inhaled, ingested, or injected). Note that the pacemaker mentioned would have included the required shielding.

This will be a very useful appliance, but probably not overpowered.

**Comment by ANickelN** [comment]
So just a note: I took some of the code from this and added it to my current save.  Everything seems to work fine and they are a nice way to have an entirely underground base without relying on solar panels.
I'm concerned about the price in regards to how common the roadblocks are and how you can essentially buy literally EVERYTHING from an entire faction if you trade even one of these at their current price of $25,000

The price should be set by looking at the pricing guide documentation, and following what it recommends.
"Rare items with considerable use tend to have the highest variety in prices" ~(1000-15000).
I agree this item has a lot of utility, and is somewhat rare, so I'm thinking around $125 should be a good fit for the post-apoc price?

Either way, this is an awesome addition and thank you for the work on it.  It has worked well in my playtesting so far.

**Comment by kevingranade** [comment]
Ditch the unshielded variant.  There's no indication that there is any amount of non functional shielding that you can remove without impairing the function of the device. Simply attempting to disassemble the device could destroy it.

**Comment by anoobindisguise** [comment]
> Ditch the unshielded variant. There's no indication that there is any amount of non functional shielding that you can remove without impairing the function of the device. Simply attempting to disassemble the device could destroy it.

My thinking was that the military put composite armor on the device to shield it from gunfire, while the "unshielded" variation is the bare-minimum NASA version that has just enough protection to be functional. But it's definitely extremely risky trying to shave armor plating off the reactor so I can go ahead and obsolete most of it + update the mapgen as the unshielded version spawns in a few places.

**Pull request merged**